src/lcm/entry_point.py

```diff
@@ -59,196 +63,116 @@ def get_lcm_function(
-    emax_calculators = []
-
-    # ==================================================================================
-    # Create stace choice space for each period
-    # ==================================================================================
-    for period in range(_mod.n_periods):
+    for period in reversed(range(internal_model.n_periods)):
         is_last_period = period == last_period
 
-        # call state space creation function, append trivial items to their lists
-        # ==============================================================================
-        state_choice_space, state_space_info = create_state_choice_space(
-            model=_mod,
+        state_choice_space = create_state_choice_space(
+            model=internal_model,
             is_last_period=is_last_period,
         )
 
-        state_choice_spaces.append(state_choice_space)
-        choice_segments.append(None)
-        state_space_infos.append(state_space_info)
-
-    # ==================================================================================
-    # Shift space info (in period t we require the space info of period t+1)
-    # ==================================================================================
-    state_space_infos = state_space_infos[1:] + [{}]  # type: ignore[list-item]
+        state_space_info = create_state_space_info(
+            model=internal_model,
+            is_last_period=is_last_period,
+        )
 
-    # ==================================================================================
-    # Create model functions
-    # ==================================================================================
-    for period in range(_mod.n_periods):
-        is_last_period = period == last_period
+        if is_last_period:
+            next_state_space_info = LastPeriodsNextStateSpaceInfo
+        else:
+            next_state_space_info = state_space_infos[period + 1]
 
-        # create the compute conditional continuation value functions and append to list
-        # ==============================================================================
         u_and_f = get_utility_and_feasibility_function(
-            model=_mod,
-            state_space_info=state_space_infos[period],
+            model=internal_model,
+            next_state_space_info=next_state_space_info,
             period=period,
             is_last_period=is_last_period,
         )
 
-        compute_ccv = create_compute_conditional_continuation_value(
+        compute_ccv = get_compute_conditional_continuation_value(
             utility_and_feasibility=u_and_f,
-            continuous_choice_variables=list(_choice_grids),
+            continuous_choice_variables=tuple(state_choice_space.continuous_choices),
         )
-        compute_ccv_functions.append(compute_ccv)
 
-        compute_ccv_argmax = create_compute_conditional_continuation_policy(
+        compute_ccp = get_compute_conditional_continuation_policy(
```

Comment:
```suggestion
        compute_cc_pol = get_compute_conditional_continuation_policy(
```
Since we have CCP estimators and there `P` stands for `probability`, this is too confusing IMO.